data_pipelines/resources/io_managers.py

```diff
@@ -68,30 +64,33 @@ def load_from_path(self, context: InputContext, path: UPath) -> xr.DataArray:
         return xr.open_dataarray(path)
 
 
-class ParquetIOManager(UPathIOManager):
+class DaskParquetIOManager(UPathIOManager):
     extension: str = ".parquet"
 
     def __init__(self, base_path: str):
         super().__init__(base_path=UPath(base_path))
 
     def dump_to_path(
-        self,
-        context: OutputContext,
-        obj: dd.DataFrame | Iterable[dd.DataFrame],
-        path: UPath,
+        self, context: OutputContext, obj: pd.DataFrame | dd.DataFrame, path: UPath
     ):
-        if isinstance(obj, dd.DataFrame):
-            obj = [obj]
-        df_iter = iter(obj)
-        first_df = next(df_iter)
-        first_df.to_parquet(path, write_index=False, overwrite=True)
-
-        for df in df_iter:
-            df.to_parquet(path, write_index=False, overwrite=False)
+        if isinstance(obj, pd.DataFrame):
+            obj.to_parquet(path)
+        else:
+            obj.to_parquet(path, overwrite=True)
 
-    def load_from_path(self, context: InputContext, path: UPath) -> dd.DataFrame:
+    def load_from_path(
+        self, context: InputContext, path: UPath | Sequence[UPath]
+    ) -> dd.DataFrame:
         return dd.read_parquet(path)
 
+    def load_input(self, context: InputContext) -> dd.DataFrame:
+        if not context.has_asset_partitions:
+            path = self._get_path(context)
+            return self._load_single_input(path, context)
+        else:
+            paths = self._get_paths_for_partitions(context)
+            return self.load_from_path(context, list(paths.values()))
```

Comment:
If I understand `context.has_asset_partitions` correctly, this would mean that a partitioned asset that uses this I/O manager would always be opened with all the partitions into a single dataframe. For the flood pipeline, this isn't an issue as the only partitioned asset which uses `DaskParquetIOManager` is `transformed_discharge` (for which we want all partitions to open together into a single dataframe), but this might not be the desired behavior for all (partitioned) assets which might use this I/O manager. Maybe the `read_all_partitions: bool = False` attribute from the old `ParquetIOManagerNew` class is a way to address this?